### PR TITLE
Devices and metric computation

### LINKED ISSUE #33: [BUG] Models are not set to evaluation mode, and are not moved to device
**Describe the bug**
Models needs to be moved to evaluation mode to have deterministic predictions, and they should be moved to the fastest available hardware, to make evaluation faster.

**To Reproduce**
Any evaluation call, e.g., `$ evaluate -m pin/senda -t sent`.

**Expected behaviour**
All models being in evaluation mode when loaded, as well as be on the correct device.

**Desktop (please complete the following information):**
 - OS: All OS'es
 - Package version [e.g. 22]: 0.0.1

**Additional context**
The data also needs to be moved to the specified device.

### PR DESCRIPTION
**Pull request by saattrupdan**
This PR implements the following:
- There is now a `get_available_devices` in `utils.py`, which returns the list of available devices, sorted from most powerful to least powerful (i.e., "cuda", "mps", "cpu").
- The `EvaluationConfig` now has `prefer_mps`, `prefer_cpu` and `device` properties, where the `device` property is set using the `get_available_devices` and the preferred devices. This is also implemented in the CLI, with `--prefer-mps` and `--prefer-cpu`.
- Changed the `Task._compute_metrics` metric to take in `predictions` and `labels` as separate arguments.
- Implemented a `Task. _prepare_predictions_and_labels` method, which defaults to not changing predictions and labels, but which allows individual tasks to do so (NER needs to do so, for instance).

This closes #33, and depends on PRs #34 and #30.

### CODE REVIEW

File: src/aiai_eval/config.py [hunk 1]
```diff
@@ -160,8 +166,35 @@ class EvaluationConfig:
...
 
+    @property
+    def device(self) -> str:
+        """The compute device to use for the evaluation.
+
+        Returns:
+            str:
+                The compute device to use for the evaluation.
+        """
+
```
Comment (AJDERS) [inline src/aiai_eval/config.py:181]:
```suggestion
if self.prefer_cpu and self.prefer_mps:
   raise InvalidEvaluation("Both '--prefer-cpu' and '--prefer-mps' where enabled.")
```

Comment (AJDERS) [reply src/aiai_eval/config.py:181]: I think it a bit unclear (at least from the users perspective) what it means to have both of these enabled? Maybe? 
Or keep the current behaviour, but throw a warning, instead?

**Review by AJDERS — CHANGES_REQUESTED** [review]

File: src/aiai_eval/config.py (hunk 1 above)
Comment (saattrupdan) [reply src/aiai_eval/config.py:165]: I changed it now to `prefer_device` instead, making it a bit more clear that these are mutually exclusive

**Review by AJDERS — APPROVED** [review]
LGTM!

**Pull request merged**